### DISCUSSION TITLE
How to make complitely new tag for botany gears?

### DISCUSSION THREAD

**Discussion by JackTalentedCoder** [space-wizards/space-station-14#25160] [Questions]
I used to work only with lua code and i have some of experience with working with code of SS14 (making prototypes of clothes and spriting) and now i want to make a complitely new tag for basic item. how to make it?

**Comment by deltanedas** [comment]
tags are defined in Resources/Prototypes/tags.yml

**Reply by JackTalentedCoder** [reply]
but how create and add one with a new mechanic?

**Reply by deltanedas** [reply]
tags dont add mechanics they are just strings (but with yml linter to prevent typos)
if you want to make code create a component and system in c#

**Reply by JackTalentedCoder** [reply]
So, i get it now, and i doing some stuff with "plantholdersystem" file in botany, and i want to do "mathematical multiplying to random count while harvesting crops". I created checking for using tag, and now i dont know how to make my idea... any ideas?

**Reply by deltanedas** [reply]
sounds like you want to have yield multiplied by a random number

in the DoHarvest method, the line `_botany.Harvest(component.Seed, user, component.YieldMod)` replace component.YieldMod with `component.YieldMod * _random.Next(minYield, maxYield)`

how you get minYield and maxYield is up to you
